module/VuFind/src/VuFind/Db/Service/SearchService.php

```diff
@@ -211,17 +225,21 @@ public function getSearchesByChecksumAndOwner(
         UserEntityInterface|int|null $userOrId = null
     ): array {
         $userId = $userOrId instanceof UserEntityInterface ? $userOrId->getId() : $userOrId;
-        $callback = function ($select) use ($checksum, $sessionId, $userId) {
-            $nest = $select->where
-                ->equalTo('checksum', $checksum)
-                ->and
-                ->nest
-                ->equalTo('session_id', $sessionId)->and->equalTo('saved', 0);
-            if (!empty($userId)) {
-                $nest->or->equalTo('user_id', $userId);
-            }
-        };
-        return iterator_to_array($this->getDbTable('search')->select($callback));
+        $dql = 'SELECT s FROM ' . $this->getEntityClass(SearchEntityInterface::class) . ' s '
+            . 'WHERE s.checksum = :checksum '
+            . 'AND s.sessionId = :sessionId '
+            . 'AND s.saved = 0';
+
+        $params = compact('checksum', 'sessionId');
+
+        if (!empty($userId)) {
+            $dql .= ' AND (s.user = :userId)';
+            $params['userId'] = $userId;
+        }
```

Comment:
I don't think the nesting has been exactly translated here. I believe this may be more faithful to the original logic:

```suggestion
        $dql = 'SELECT s FROM ' . $this->getEntityClass(SearchEntityInterface::class) . ' s '
            . 'WHERE s.checksum = :checksum AND ';
        $extraClauses = ['(s.sessionId = :sessionId AND s.saved = 0)'];

        $params = compact('checksum', 'sessionId');

        if (!empty($userId)) {
            $extraClauses[] = 's.user = :userId';
            $params['userId'] = $userId;
        }
        
        $dql .= '(' . implode(' OR ', $extraClauses) . ')';
```